mcs/class/referencesource/mscorlib/system/rttype.cs

```diff
@@ -3414,8 +3414,14 @@ public override Type GetInterface(String fullname, bool ignoreCase)
 #if MONO
             List<RuntimeType> list = null;
             foreach (RuntimeType t in GetInterfaces ()) {
-                if (t.Name != name)
-                    continue;
+
+                if (ignoreCase) {
+                    if (t.Name.ToLowerInvariant() != name)
```

Comment:
shouldn't it be `t.Name.ToLowerInvariant () != name.ToLowerInvariant`?

Reply:
Better to use special string comparer, see https://msdn.microsoft.com/en-us/library/e6883c06(v=vs.110).aspx

Reply:
@lewurm RuntimeType.FilterHelper changes `name` to lower case.

Reply:
I think we could simplify this to:

```csharp
if (!t.Name.Equals(name, ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
  continue;
```